Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -568,6 +578,16 @@ define([
             }
             that._credit = credit;
 
+            if (defined(properties.customTags)) {
+                for (var tag in properties.customTags) {
+                    if (properties.customTags.hasOwnProperty(tag)) {
+                        var targetTag = '{' + tag + '}';
+                        tags[targetTag] = properties.customTags[tag]; //eslint-disable-line no-use-before-define
```

Comment:
I just realized there is a major issue here.  Since `tags` and `pickFeaturesTags` used to be readonly, the same set is shared across multiple instances.  However, now we are modifying `tags` and `pickFeaturesTags` which means that the modified version is used by all instances.  Shouldn't these objects be per-instance now?

Reply:
The `tags` and `pickFeaturesTags` are the same for all objects.
Maybe I can just append the `customTags` values to `tags` values in a new object and pass this new object to  `urlTemplateToParts`.